pkg/hiveregistrator/resources.go

```diff
@@ -0,0 +1,98 @@
+package hiveregistrator
+
+import (
+	"encoding/base64"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/apis/hive/v1/azure"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ClusterNamespace(w WorkloadCluster, namespace string) *corev1.Namespace {
+
+	return &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ClusterNamespaceName(w),
+			//Annotations: map[string]string{projectv1.ProjectNodeSelector: ""},
+		},
+	}
+
+}
+
+func KubeAdminSecret(w WorkloadCluster, secretName string, namespace string) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			"kubeconfig":     []byte(w.KubeConfig),
+			"raw-kubeconfig": []byte(w.KubeConfig),
```

Comment:
Why do we need `raw-kubeconfig`? I don't remember doing anything like this while practicing adoption.

Reply:
Dropping raw-kubeconfig. Works fine without.